### PR TITLE
update block validator

### PR DESCRIPTION
**Pull request by turuslan**
- remove power table and old block validator
- update block validator
- move and reuse some code
- `rand[31] &= 0x3f` like lotus
- `static inline` bug
- test

### CODE REVIEW

File: core/primitives/block/eligible.hpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#pragma once
+
+#include "cbor_blake/ipld_version.hpp"
+#include "primitives/tipset/tipset.hpp"
+#include "vm/actor/builtin/states/miner/miner_actor_state.hpp"
+#include "vm/actor/builtin/states/storage_power/storage_power_actor_state.hpp"
+#include "vm/actor/builtin/types/storage_power/policy.hpp"
+#include "vm/state/impl/state_tree_impl.hpp"
+
+namespace fc::miner_eligible_to_miner {
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/block/eligible.hpp:15]: I think the right place for file is in `core/blockchain/block_validator` and namespace 'blockchain::block_validator`.

File: core/primitives/block/win_sectors.hpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "proofs/impl/proof_engine_impl.hpp"
+#include "vm/actor/builtin/states/miner/miner_actor_state.hpp"
+
+namespace fc::get_sectors_for_winning_post {
```
Comment (Alexey-N-Chernyshov) [inline core/primitives/block/win_sectors.hpp:11]: The same, it is not a primitive.

File: core/blockchain/block_validator/validator.cpp [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+          OUTCOME_TRY(wmeta.secp_messages.append(cid));
+          return outcome::success();
+        }));
+    OUTCOME_TRY(root, setCbor(null_ipld, wmeta));
+    if (root != block.messages) {
+      return ERROR_TEXT("validateMessages: wrong root");
+    }
+    return outcome::success();
+  }
+}  // namespace fc::blockchain::block_validator
```
Comment (Elestrias) [inline core/blockchain/block_validator/validator.cpp:272]: empty line at the end of a file

File: core/blockchain/block_validator/validator.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    BlockValidator(MapPtr kv, const EnvironmentContext &envx);
+
+    outcome::result<void> validate(const TsBranchPtr &branch,
+                                   const BlockHeader &block);
+    outcome::result<void> validateMessages(const BlockHeader &block,
+                                           const TipsetCPtr &ts,
+                                           StateTreeImpl &tree);
+  };
+}  // namespace fc::blockchain::block_validator
```
Comment (Elestrias) [inline core/blockchain/block_validator/validator.hpp:36]: empty line

File: core/miner/mining.hpp [hunk 5]
```diff
@@ -67,9 +67,5 @@ namespace fc::mining {
     boost::optional<MiningBaseInfo> info;
   };
 
-  int64_t computeWinCount(BytesIn ticket,
-                          const BigInt &power,
-                          const BigInt &total_power);
-
   double ticketQuality(BytesIn ticket);
 }  // namespace fc::mining
```
Comment (Elestrias) [inline core/miner/mining.hpp:71]: Empty line

File: core/primitives/block/eligible.hpp [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    if (parent->height() <= parent_miner_info->consensus_fault_elapsed) {
+      return false;
+    }
+    return true;
+  }
+}  // namespace fc::miner_eligible_to_miner
+
+namespace fc {
+  using miner_eligible_to_miner::minerEligibleToMine;
+}  // namespace fc
```
Comment (Elestrias) [inline core/primitives/block/eligible.hpp:79]: empty line

File: core/primitives/block/rand.hpp [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        worker, data, *block.block_sig);
+  }
+}  // namespace fc::primitives::block
+
+namespace fc {
+  using primitives::block::blakeBigInt;
+  using primitives::block::BlockRand;
+  using primitives::block::checkBlockSignature;
+  using primitives::block::computeWinCount;
+}  // namespace fc
```
Comment (Elestrias) [inline core/primitives/block/rand.hpp:96]: empty line

File: core/primitives/block/win_sectors.hpp [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        });
+      }
+    }
+    return sectors;
+  }
+}  // namespace fc::get_sectors_for_winning_post
+
+namespace fc {
+  using get_sectors_for_winning_post::getSectorsForWinningPoSt;
+}  // namespace fc
```
Comment (Elestrias) [inline core/primitives/block/win_sectors.hpp:70]: empty line

File: core/vm/message/valid.hpp [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    if (msg.gas_limit > kBlockGasLimit) {
+      return false;
+    }
+    if (msg.gas_limit < gas) {
+      return false;
+    }
+    return true;
+  }
+}  // namespace fc::vm::message
```
Comment (Elestrias) [inline core/vm/message/valid.hpp:44]: empty line

File: test/core/blockchain/block_validator_test.cpp [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    ts = head;
+    while (ts->height() != 0) {
+      for (const auto &block : ts->blks) {
+        validator.validate(ts_main, block).value();
+      }
+      ts = envx.ts_load->load(ts->getParents()).value();
+    }
+  }
+}  // namespace fc::blockchain::block_validator
```
Comment (Elestrias) [inline test/core/blockchain/block_validator_test.cpp:71]: empty line

File: core/storage/car/cids_index/cids_index.hpp [hunk 11]
```diff
@@ -63,9 +63,9 @@ namespace fc::storage::cids_index {
     return memcmp(&l.key, &r, sizeof(CbCid)) < 0;
   }
 
-  static inline const Row kHeaderV0{
+  inline const Row kHeaderV0{
```
Comment (Elestrias) [inline core/storage/car/cids_index/cids_index.hpp:66]: Why it was removed from static state, why do we need to instantiate the class to get access to constant that is used in higher levels of program call trace ?

File: core/proofs/impl/proof_engine_impl.cpp [hunk 12]
```diff
@@ -38,6 +38,11 @@ namespace fc::proofs {
   using primitives::sector::RegisteredAggregationProof;
   using sector_storage::zerocomm::getZeroPieceCommitment;
 
+  inline auto rand31(PoStRandomness rand) {
+    rand[31] &= 0x3f;
+    return rand;
+  }
+
```
Comment (Elestrias) [inline core/proofs/impl/proof_engine_impl.cpp:45]: If it is an operation under the PoStRandomness, why not to modify it via inner class function f.e. self modified getter

File: core/storage/car/cids_index/cids_index.hpp (hunk 11 above)
Comment (turuslan) [reply core/storage/car/cids_index/cids_index.hpp:66]: [`static` linkage](https://en.cppreference.com/w/cpp/language/storage_duration) != [`static` member](https://en.cppreference.com/w/cpp/language/static)

File: core/proofs/impl/proof_engine_impl.cpp (hunk 12 above)
Comment (turuslan) [reply core/proofs/impl/proof_engine_impl.cpp:45]: This function is only used in this file.  
Currently `PoStRandomness` is alias to 32 bytes, so we can't add member function.

File: core/common/git_commit_version/.gitignore [hunk 13]
```diff
@@ -0,0 +1 @@
+git_commit_version.cpp
```
Comment (wer1st) [inline core/common/git_commit_version/.gitignore:1]: For what?

**Review by wer1st — APPROVED** [review]

File: core/common/git_commit_version/.gitignore (hunk 13 above)
Comment (turuslan) [reply core/common/git_commit_version/.gitignore:1]: because it's a generated file

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**